explainability-service/src/main/java/org/kie/trustyai/service/prometheus/PrometheusScheduler.java

```diff
@@ -101,6 +101,11 @@ public void register(String metricName, UUID id, BaseMetricRequest request) {
         requests.get(metricName).put(id, request);
     }
 
+    public void delete(String metricName, UUID id) {
+        getRequests(metricName).remove(id);
+        this.getPublisher().removeGauge(metricName, id);
```

Comment:
Similar comment here, to enforce proper behaviour we should make `getPublisher` private/proteced and expose only the necessary wrapped methods.

Same here: fine for me to postpone to a different ticket

Reply:
done